### PR TITLE
Static probes

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Make the probes predictable for timing tests

### Motivation and Context

depending on implementation, static probes may give clearer signal about presence of a sidechannel

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [x] NSS
  - [x] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)

### CODE REVIEW

File: scripts/test-bleichenbacher-timing.py [hunk 1]
```diff
@@ -149,6 +154,8 @@ def main():
             no_sni = True
         elif opt == "--cpu-list":
             affinity = arg
+        elif opt == "--static-end":
```
Comment (tomato42) [inline scripts/test-bleichenbacher-timing.py:157]: s/end/enc/

**Review by ansasaki — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r4, 2 of 2 files at r5, 3 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)* status: all files reviewed, 3 unresolved discussions (waiting on @tomato42)

___
*[scripts/test-bleichenbacher-timing.py, line 78 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthOMU7-WpB8CP9VaA:-MKthOMV63ahik8e87M2:bvpxx0q) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/scripts/test-bleichenbacher-timing.py#L78)):*
> ```Python
>     print("                See taskset(1) man page for the syntax of this")
>     print("                option. Not used by default.")
>     print(" --static-enc   Re-use once generated RSA cipihertext. This may make the")
> ```

s/cipihertext/ciphertext/
___
*[tests/test_tlsfuzzer_messages.py, line 880 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthmDH51krecA5rYEg:-MKthmDH51krecA5rYEh:b76l05x) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/tests/test_tlsfuzzer_messages.py#L880)):*
> ```Python
>         self.assertEqual(decrypt[2:], bytearray([1]*8))
> 
>     def test_generate_with_reuse(self):
> ```

This name is a bit confusing for me as the test is not reusing the encrypted pre-master secret.

**Review by tomato42 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)* status: all files reviewed, 3 unresolved discussions (waiting on @ansasaki and @tomato42)

___
*[tests/test_tlsfuzzer_messages.py, line 880 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthmDH51krecA5rYEg:-MKtoFhAA450gETrKr8Y:bi9jjix) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/tests/test_tlsfuzzer_messages.py#L880)):*
<details><summary><i>Previously, ansasaki (Anderson Sasaki) wrote…</i></summary><blockquote>

This name is a bit confusing for me as the test is not reusing the encrypted pre-master secret.
</blockquote></details>

it reuses the generator node, any suggestion for a new name?

**Review by ansasaki — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)* status: all files reviewed, 3 unresolved discussions (waiting on @tomato42)

___
*[tests/test_tlsfuzzer_messages.py, line 880 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthmDH51krecA5rYEg:-MKtpJcyCnZOMVQ3npvU:biv8yqf) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/tests/test_tlsfuzzer_messages.py#L880)):*
<details><summary><i>Previously, tomato42 (Hubert Kario) wrote…</i></summary><blockquote>

it reuses the generator node, any suggestion for a new name?
</blockquote></details>

Hum, maybe explicitly spell out what is being reused, like ``test_generate_with_node_reuse()``? In any case, feel free to ignore my comment if it is clear to you (I didn't check other files to see if this is a well known naming pattern).

**Review by tomato42 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @ansasaki)

___
*[scripts/test-bleichenbacher-timing.py, line 78 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthOMU7-WpB8CP9VaA:-MKtyHi445l2y_yOVzG4:b-896fix) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/scripts/test-bleichenbacher-timing.py#L78)):*
<details><summary><i>Previously, ansasaki (Anderson Sasaki) wrote…</i></summary><blockquote>

s/cipihertext/ciphertext/
</blockquote></details>

Done.
___
*[tests/test_tlsfuzzer_messages.py, line 880 at r6](https://reviewable.io/reviews/tomato42/tlsfuzzer/715#-MKthmDH51krecA5rYEg:-MKtyLl6BlxF-wWD0LWS:b-dx7u4u) ([raw file](https://github.com/tomato42/tlsfuzzer/blob/934e7352a51ead98f31b8598815f138a3d39b2b6/tests/test_tlsfuzzer_messages.py#L880)):*
<details><summary><i>Previously, ansasaki (Anderson Sasaki) wrote…</i></summary><blockquote>

Hum, maybe explicitly spell out what is being reused, like ``test_generate_with_node_reuse()``? In any case, feel free to ignore my comment if it is clear to you (I didn't check other files to see if this is a well known naming pattern).
</blockquote></details>

if it's not understandable for you, it won't be understandable to me in two months, fixed

**Review by ansasaki — APPROVED** [review]
> Reviewed 5 of 5 files at r7, 3 of 3 files at r8, 1 of 1 files at r9.
> *[Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/715)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**